### DISCUSSION TITLE
Undo + Copy & Paste

### DISCUSSION THREAD

**Discussion by ghost** [talmolab/sleap#883] [Ideas]
Hi
It would be great if we have undo (CTRL+Z) option. 
Moreover, it is very necessary to have the Copy & Pase option for the labeled skeleton in SLEAP for a new video (like the Add Instance from prior frame option in the same video, we could add instances from another project or video). We have many keypoints in our project and it would be convenient to be able to copy and paste or even import the labeled skeleton from another project. Now, whenever we create a new project, we have randomly positioned keypoints and we have to reorder all of them. Even if we have multiple videos in the project and in a new video (same project) we can add instances from other videos, it would be very handy.

Thanks

**Comment by talmo** [comment]
Hi @aarahvar2,

Thanks, these are good suggestions. We started working on this a while back but it's a little tricky to implement Undo correctly.

Copy-paste should be a bit easier though, so maybe we can think of a way to implement that more quickly.

We'll update this thread when we start working on either of these features.

Cheers,

Talmo

**Reply by ghost** [reply]
Thanks Talmo. They will make Sleap like standard software.

**Comment by talmo** [comment]
Forgot to update this, but we implemented copy-pasting in https://github.com/talmolab/sleap/pull/1206! It is available on SLEAP v1.3.0 (currently in pre-release).
